lighthouse-core/audits/dobetterweb/uses-http2.js

```diff
@@ -43,7 +43,7 @@ class UsesHTTP2Audit extends Audit {
 
       // Filter requests that are on the same host as the page and not over h2.
       const resources = networkRecords.filter(record => {
-        const requestHost = new URL(record._url).host;
+        const requestHost = record._parsedURL && record._parsedURL.host;
```

Comment:
netrequest.parsedURL uses a different implementation of url parsing. (Though I hope that wouldn't matter).

Alternatively we could add caching to our URL shim? That'd have the benefit of improving perf for all audits/gatherers.

Don't feel strongly, just raising the idea.